tests/gadget-zero/usb-gadget0.c

```diff
@@ -181,37 +184,60 @@ static struct {
 	.pattern_counter = 0,
 };
 
+/* aligned / unaligned test flag.
+ * Using a bool type for this instead, in some cases, can cause gcc to place other critical buffers
+ * at unaligned addresses ! like "usbd_control_buffer".
+ * If this is 0 (default), use 16-bit aligned buffers.
+ */
+static int test_unaligned = 0;
+
 static void gadget0_ss_out_cb(usbd_device *usbd_dev, uint8_t ep)
 {
 	(void) ep;
+	uint16_t x;
 	/* TODO - if you're really keen, perf test this. tiva implies it matters */
 	/* char buf[64] __attribute__ ((aligned(4))); */
-	char buf[64];
+	uint16_t buf[BULK_EP_MAXPACKET / 2 + 1];
+	uint8_t *dest;
+
 	trace_send_blocking8(0, 'O');
-	uint16_t x = usbd_ep_read_packet(usbd_dev, ep, buf, sizeof(buf));
+	if (test_unaligned) {
+		dest = ((uint8_t *) buf) + 1;
+	} else {
+		dest = (uint8_t *) buf;
+	}
+	x = usbd_ep_read_packet(usbd_dev, ep, dest, BULK_EP_MAXPACKET);
 	trace_send_blocking8(1, x);
 }
 
 static void gadget0_ss_in_cb(usbd_device *usbd_dev, uint8_t ep)
 {
 	(void) usbd_dev;
 	trace_send_blocking8(0, 'I');
-	uint8_t buf[64];
+	uint16_t buf[BULK_EP_MAXPACKET / 2 + 1];
+	uint8_t *src;
+
+	if (test_unaligned) {
+		src = ((uint8_t *) buf) + 1;
```

Comment:
how do you know that it was aligned before? is that because you used a uint16_t buf that must get aligned?  Why not just use __attribute__(aligned(4)) on the uint8_t buffer, and +1 there?  more obvious what's going on?